middle_end/flambda2/simplify/simplify_binary_primitive.ml

```diff
@@ -1153,16 +1214,28 @@ let simplify_binary_primitive dacc (prim : P.binary_primitive) ~arg1 ~arg1_ty
     | Float_arith op -> Binary_float_arith.simplify op
     | Float_comp op -> Binary_float_comp.simplify op
     | Phys_equal (kind, op) -> simplify_phys_equal op kind
-    | Block_load _ | String_or_bigstring_load _ | Bigarray_load _ ->
+    | Block_load _ ->
```

Comment:
If I read the code correctly, `Immutable` loads are already handled earlier, so here we only handle `Mutable` loads. I don't think we need to record projections for mutable loads, as all symbols are supposed to be immutable anyway.